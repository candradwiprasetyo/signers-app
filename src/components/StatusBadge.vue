<template>
  <div>
    <span
      :class="[
        'material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto',
        active ? 'bg-green-400' : 'bg-red-400',
      ]"
      :title="active ? 'Aktif' : 'Tidak Aktif'"
    >
      {{ active ? 'check' : 'close' }}
    </span>
  </div>
</template>

<script setup lang="ts">
defineProps({
  active: Boolean,
})
</script>
