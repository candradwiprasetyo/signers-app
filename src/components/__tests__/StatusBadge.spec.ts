import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import StatusBadge from '../StatusBadge.vue'

describe('StatusBadge', () => {
  it('render active icon when active is true', () => {
    const wrapper = mount(StatusBadge, {
      props: {
        active: true,
      },
    })

    const span = wrapper.find('span')
    expect(span.text()).toBe('check')
    expect(span.attributes('title')).toBe('Aktif')
    expect(span.classes()).toContain('bg-green-400')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render non active icon when active is false', () => {
    const wrapper = mount(StatusBadge, {
      props: {
        active: false,
      },
    })

    const span = wrapper.find('span')
    expect(span.text()).toBe('close')
    expect(span.attributes('title')).toBe('Tidak Aktif')
    expect(span.classes()).toContain('bg-red-400')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
