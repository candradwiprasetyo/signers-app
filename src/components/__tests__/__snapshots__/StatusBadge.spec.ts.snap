// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusBadge > render active icon when active is true 1`] = `"<div><span class="material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto bg-green-400" title="Aktif">check</span></div>"`;

exports[`StatusBadge > render non active icon when active is false 1`] = `"<div><span class="material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto bg-red-400" title="Tidak Aktif">close</span></div>"`;
